new_platforms/src/optee_common.c

```diff
@@ -7,17 +7,274 @@
+
+        if (end_ext_octet != *p + len)
+            return (
+                MBEDTLS_ERR_X509_INVALID_EXTENSIONS +
+                MBEDTLS_ERR_ASN1_LENGTH_MISMATCH);
+
+        *ClaimOidData = *p;
+        *ClaimOidDataLen = len;
+        result = OE_OK;
+        break;
+    }
+
+    return result;
+}
+
+static oe_result_t get_claims(
+    mbedtls_x509_crt *cert,
+    uint8_t *signer_id_buffer,
+    size_t signed_id_buffer_size,
+    uint8_t* unique_id_buffer,
+    size_t unique_id_buffer_size)
+{
+    uint8_t* claimOidData;
+    size_t claimOidDataLen;
+    oe_result_t result = GetClaimOidData(
+        cert->v3_ext.p, cert->v3_ext.len, &claimOidData, &claimOidDataLen);
+    if (result != OE_OK)
+    {
+        goto Cleanup;
+    }
+
+    const uint8_t* valueBuffer;
+    size_t valueBufferLen;
+    RIOT_STATUS status = GetClaim(
+        claimOidData,
+        claimOidDataLen,
+        TCPS_IDENTITY_MAP_AUTH,
+        &valueBuffer,
+        &valueBufferLen);
+    if (status != RIOT_SUCCESS)
+    {
+        result = OE_FAILURE;
+        goto Cleanup;
+    }
+
+    status = RiotCrypt_Hash(
+        signer_id_buffer,
+        signed_id_buffer_size,
+        valueBuffer,
+        valueBufferLen);
+    if (status != RIOT_SUCCESS)
+    {
+        result = OE_FAILURE;
+        goto Cleanup;
+    }
+
+    status = GetClaim(
+        claimOidData,
+        claimOidDataLen,
+        TCPS_IDENTITY_MAP_FWID,
+        &valueBuffer,
+        &valueBufferLen);
+    if (status == RIOT_SUCCESS)
+    {
+        if (valueBufferLen != unique_id_buffer_size)
+        {
+            result = OE_FAILURE;
+            goto Cleanup;
+        }
+
+        memcpy(unique_id_buffer, valueBuffer, valueBufferLen);
+    }
+    else
+    {
+        status = GetClaim(
+            claimOidData,
+            claimOidDataLen,
+            TCPS_IDENTITY_MAP_PUBKEY,
+            &valueBuffer,
+            &valueBufferLen);
+        if (status != RIOT_SUCCESS)
+        {
+            result = OE_FAILURE;
+            goto Cleanup;
+        }
+
+        status = RiotCrypt_Hash(
+            unique_id_buffer,
+            unique_id_buffer_size,
+            valueBuffer,
+            valueBufferLen);
+        if (status != RIOT_SUCCESS)
+        {
+            result = OE_FAILURE;
+            goto Cleanup;
+        }
+    }
+
+Cleanup:
+    return result;
+}
+
 oe_result_t oe_parse_report(
     _In_reads_bytes_(report_size) const uint8_t* report,
     _In_ size_t report_size,
     _Out_ oe_report_t* parsed_report)
 {
-    /* Not yet supported */
-    return OE_UNSUPPORTED;
+    oe_result_t result = OE_OK;
+    
+    if (report == NULL || report_size == 0 || parsed_report == NULL)
+    {
+        return OE_INVALID_PARAMETER;
+    }
+
+    parsed_report->size = sizeof(*parsed_report);
+    parsed_report->type = OE_ENCLAVE_TYPE_TRUSTZONE;
+
+    parsed_report->enclave_report_size = report_size;
+    parsed_report->enclave_report = (uint8_t*)&report;
```

Comment:
the & here looks like a bug since it's the address of a stack variable.

Reply:
Thanks, done.